os/arch/arm/src/amebasmart/amebasmart_i2s.c

```diff
@@ -594,6 +594,9 @@ static uint32_t i2s_txdatawidth(struct i2s_dev_s *dev, int bits)
 	/* Support 16, 24, 32 bits */
 	DEBUGASSERT(priv && (bits == I2S_BITS_PER_SAMPLE_16BIT || bits == I2S_BITS_PER_SAMPLE_32BIT || bits == I2S_BITS_PER_SAMPLE_24BIT));
 
+	/* Unregister Port direction */
+	i2s_deinit(&priv->i2s_object);
```

Comment:
Hello @kaikaikaime , I think this should be as follows. Could you please confirm? I am getting a crash if I run the current code.
```suggestion
	i2s_deinit(priv->i2s_object);
```
Also even after changing it as above, it doesn't seem to work. As soon as deinit is called, uart logs are getting corrupted.

Reply:
Hi abhishek, I have remove ampersasnd and disable of the clock which interferes with other peripherals